packages/myst-transforms/src/math.ts

```diff
@@ -241,6 +241,30 @@ export function mathNestingTransform(
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   file: VFile,
 ) {
+  const paragraphs = selectAll('paragraph', tree) as GenericParent[];
+  paragraphs.forEach((paragraph) => {
+    if (paragraph.children.length === 1) return; // no need to traverse, the math node can never be tight!
+    paragraph.children.forEach((child, index) => {
```

Comment:
So for this, math inside a paragraph is _always_ tight? Is there some redundancy with this and the parse-level logic? Like, if we get to this point, the node probably already has correct `tight` value since in the source the lines above/below aren't empty... but _maybe_ not and this cleans it up...?

Reply:
Oh, this is to handle `dollar math` specifically. (AMS and directive math are handled at parse, dollar math is handled here) 👍

Reply:
Adding a comment!